src/main/java/cz/muni/fi/pa165/bluebat/entity/Excursion.java

```diff
@@ -1,5 +1,121 @@
 package cz.muni.fi.pa165.bluebat.entity;
 
+import javax.persistence.*;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by Tomáš Hampl on 3.3.21.
+ */
+
+@Entity
 public class Excursion {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+
+    @Column(nullable = false)
+    private LocalDate dateFrom;
+
+    @Column(nullable = false)
+    private Timestamp duration;
+
+    private String destination;
+
+    private String description;
+
+    @OneToMany()
+    @OrderBy("validFrom DESC")
+    @JoinTable(name = "EXCURSION_PRICE")
+    private List<Price> prices = new ArrayList<>();
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getDateFrom() {
+        return dateFrom;
+    }
+
+    public void setDateFrom(LocalDate dateFrom) {
+        this.dateFrom = dateFrom;
+    }
+
+    public Timestamp getDuration() {
+        return duration;
+    }
+
+    public void setDuration(Timestamp duration) {
+        this.duration = duration;
+    }
+
+    public String getDestination() {
+        return destination;
+    }
+
+    public void setDestination(String destination) {
+        this.destination = destination;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
```

Comment:
Zde by šel pro generování getterů a setterů použít Lombok. Všechen tento kód by se pak mohl odstranit. Anotace Getter (případně Setter) umístěná nad classou jej vygeneruje pro všechny atributy. Pokud se anotace umístí nad atribut, pak je kod vygenerován pouze pro daný atribut. Protože to však ve výsledku generuje stejný kod jako IDE, pokud chceme tyto metody nějak modifikovat (což zde platí pro getPrices), Lombok nelze pro daný atribut použít. Jde s nim vygenerovat i toString, equals a hashCode, nicméně pro equals a hashCode zde opět nelze použít, protože je potřeba tyto metody mírně upravit oproti tomu, jak je IDE vygeneruje.